airflow/models.py

```diff
@@ -1367,10 +1367,12 @@ def append_only_new(self, l, item):
             l.append(item)
 
     def _set_relatives(self, task_or_task_list, upstream=False):
-        if isinstance(task_or_task_list, BaseOperator):
-            task_or_task_list = [task_or_task_list]
-        for task in task_or_task_list:
-            if not isinstance(task_or_task_list, list):
+        try:
+            task_list = list(task_or_task_list)
```

Comment:
I can change it to that if you prefer, however it doesn't allow you to have custom classes where __iter__ is defined to be passed in here. Would a comment alleviate your concerns?